.idea/checkstyle-idea.xml

```diff
@@ -3,6 +3,7 @@
   <component name="CheckStyle-IDEA" serialisationVersion="2">
     <checkstyleVersion>10.18.0</checkstyleVersion>
     <scanScope>JavaOnly</scanScope>
+    <scanBeforeCheckin>true</scanBeforeCheckin>
```

Comment:
what does this do? I can't find documentation on this

Reply:
It means Checkstyle automatically runs a code check before you "check in"/add your code to Git.